lib/galaxy/model/metadata.py

```diff
@@ -682,15 +682,16 @@ def new_file(self, dataset=None, **kwds):
             # we need to make a tmp file that is accessable to the head node,
             # we will be copying its contents into the MetadataFile objects filename after restoring from JSON
             # we do not include 'dataset' in the kwds passed, as from_JSON_value() will handle this for us
-            return MetadataTempFile(**kwds)
+            return MetadataTempFile(metadata_tmp_files_dir=metadata_tmp_files_dir, **kwds)
 
 
 # This class is used when a database file connection is not available
 class MetadataTempFile:
     tmp_dir = "database/tmp"  # this should be overwritten as necessary in calling scripts
 
-    def __init__(self, **kwds):
+    def __init__(self, metadata_tmp_files_dir=None, **kwds):
         self.kwds = kwds
+        self.tmp_dir = metadata_tmp_files_dir
```

Comment:
Ah, that's a good point. We do need this for legacy datatypes, so I think it should be
```suggestion
        if metadata_tmp_files_dir:
            self.tmp_dir = metadata_tmp_files_dir
```

Reply:
That should work, I think.